packages/atomic/src/components/search/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -71,6 +71,7 @@ export class AtomicSmartSnippetSource implements InitializableComponent {
           <atomic-result-text
             field="title"
             default="no-title"
+            key={this.source.uniqueId}
```

Comment:
Some logic was already added in the result list to ensure results are re-rendered when their uniqueId is changed.

I'm was a bit tempted to introduce a result provider component that would handle re-rendering, but it would be overkill for this PR.